tests/t221-elemrestriction.c

```diff
@@ -0,0 +1,69 @@
+/// @file
+/// Test creation, use, and destruction of an element restriction oriented with unsigned application
+/// \test Test creation, use, and destruction of an element restriction oriented with unsigned application
+#include <ceed.h>
+#include <ceed/backend.h>
+#include <math.h>
+#include <stdio.h>
+
+int main(int argc, char **argv) {
+  Ceed                ceed;
+  CeedVector          x, y_oriented, y_unsigned;
+  CeedInt             num_elem = 6, p = 2, dim = 1;
+  CeedInt             ind[p * num_elem];
+  bool                orient[p * num_elem];
+  CeedScalar          x_array[num_elem + 1];
+  CeedElemRestriction elem_restriction;
+
+  CeedInit(argv[1], &ceed);
+
+  CeedVectorCreate(ceed, num_elem + 1, &x);
+  for (CeedInt i = 0; i < num_elem + 1; i++) x_array[i] = 10 + i;
+  CeedVectorSetArray(x, CEED_MEM_HOST, CEED_USE_POINTER, x_array);
+  CeedVectorCreate(ceed, num_elem * 2, &y_oriented);
+  CeedVectorCreate(ceed, num_elem * 2, &y_unsigned);
+
+  for (CeedInt i = 0; i < num_elem; i++) {
+    ind[2 * i + 0] = i;
+    ind[2 * i + 1] = i + 1;
+    // flip the dofs on element 1,3,...
+    orient[2 * i + 0] = (i % (2)) * -1 < 0;
+    orient[2 * i + 1] = (i % (2)) * -1 < 0;
+  }
+  CeedElemRestrictionCreateOriented(ceed, num_elem, p, dim, 1, num_elem + 1, CEED_MEM_HOST, CEED_USE_POINTER, ind, orient, &elem_restriction);
+
+  CeedElemRestrictionApply(elem_restriction, CEED_NOTRANSPOSE, x, y_oriented, CEED_REQUEST_IMMEDIATE);
+  CeedElemRestrictionApplyUnsigned(elem_restriction, CEED_NOTRANSPOSE, x, y_unsigned, CEED_REQUEST_IMMEDIATE);
```

Comment:
My only comment before approving is that this test could also check `CeedElemRestrictionCreateUnsignedCopy` by copying the original `elem_restriction` and ensuring the result of `CeedElemRestrictionApply` matches the unsigned apply? If that is superfluous just say so and I will approve.

Reply:
I'm a fan of belt and suspenders! Done